pkg/api/controllers/account_controller.go

```diff
@@ -123,20 +127,35 @@ func (ctl *AccountController) GetAccount(c *gin.Context) {
 		return
 	}
 
+	if acc.Address != c.Param("address") {
+		ResponseError(c, ledger.NewNotFoundError("unknown account"))
+		return
+	}
+
 	respondWithData[core.Account](c, http.StatusOK, acc)
 }
 
 func (ctl *AccountController) PostAccountMetadata(c *gin.Context) {
 	l, _ := c.Get("ledger")
 
-	var m core.Metadata
-	if err := c.ShouldBindJSON(&m); err != nil {
+	if !core.ValidateAddress(c.Param("address")) {
+		ResponseError(c, ledger.NewValidationError("invalid address"))
+		return
+	}
+
+	acc, err := l.(*ledger.Ledger).GetAccount(c.Request.Context(), c.Param("address"))
+	if err != nil {
 		ResponseError(c, err)
 		return
 	}
+	if acc.Address != c.Param("address") {
```

Comment:
Should not happen.

Reply:
I also changed it to:
`if acc.Address == "" {`
Is it more clear?